Modules/_sqlite/cursor.c

```diff
@@ -506,6 +515,245 @@ stmt_step(sqlite3_stmt *statement)
     return rc;
 }
 
+static int
+bind_param(pysqlite_Statement *self, int pos, PyObject *parameter)
+{
+    int rc = SQLITE_OK;
+    const char *string;
+    Py_ssize_t buflen;
+    parameter_type paramtype;
+
+    if (parameter == Py_None) {
+        rc = sqlite3_bind_null(self->st, pos);
+        goto final;
+    }
+
+    if (PyLong_CheckExact(parameter)) {
```

Comment:
I know you just moved this code, but it's weird that we do CheckExact() and then Check() immediately after. Is that a micro-optimization?

Reply:
I remember discussing this piece of code with Serhiy a year ago or two, and IIRC, it was intended to be a kind of a micro-optimisation. I'm not sure it is worth it, though. I'll see if I can dig up that old conversation.

Reply:
Don't worry about it then, not worth spending too much time on.

Reply:
Too late ;) gh-87433

Reply:
Thanks, I think I agree with Serhyi there: probably not worth applying this optimization if someone proposes it now, but there's little gain in removing it since it's already there.

Reply:
Yes, I agree, for now at least.